crates/sui-framework/packages/sui-framework/sources/crypto/bls12381.move

```diff
@@ -18,4 +23,221 @@ module sui::bls12381 {
+    }
+
+    public fun g1_add(e1: &Element<G1>, e2: &Element<G1>): Element<G1> {
+        group_ops::add(G1_TYPE, e1, e2)
+    }
+
+    public fun g1_sub(e1: &Element<G1>, e2: &Element<G1>): Element<G1> {
+        group_ops::sub(G1_TYPE, e1, e2)
+    }
+
+    public fun g1_mul(e1: &Element<Scalar>, e2: &Element<G1>): Element<G1> {
+        group_ops::mul(G1_TYPE, e1, e2)
+    }
+
+    // Returns e2 / e1, fails if scalar is zero.
+    public fun g1_div(e1: &Element<Scalar>, e2: &Element<G1>): Element<G1> {
+        group_ops::div(G1_TYPE, e1, e2)
+    }
+
+    public fun g1_neg(e: &Element<G1>): Element<G1> {
+        g1_sub(&g1_identity(), e)
+    }
+
+    // Hash using DST = BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_
+    public fun hash_to_g1(m: &vector<u8>): Element<G1> {
+        group_ops::hash_to(G1_TYPE, m)
+    }
+
+    // Let 'scalars' be the vector [s1, s2, ..., sn] and 'elements' be the vector [e1, e2, ..., en].
+    // Returns s1*e1 + s2*e2 + ... + sn*en.
+    public fun g1_multi_scalar_multiplication(scalars: &vector<Element<Scalar>>, elements: &vector<Element<G1>>): Element<G1> {
+        group_ops::multi_scalar_multiplication(G1_TYPE, scalars, elements)
+    }
+
+    /////////////////////////////////
+    ////// G2 group operations //////
+
+    public fun g2_from_bytes(bytes: &vector<u8>): Element<G2> {
+        group_ops::from_bytes(G2_TYPE, bytes, false)
+    }
+
+    public fun g2_identity(): Element<G2> {
+        group_ops::from_bytes(G2_TYPE, &G2_IDENTITY_BYTES, true)
+    }
+
+    public fun g2_generator(): Element<G2> {
+        group_ops::from_bytes(G2_TYPE, &G2_GENERATOR_BYTES, true)
+    }
+
+    public fun g2_add(e1: &Element<G2>, e2: &Element<G2>): Element<G2> {
+        group_ops::add(G2_TYPE, e1, e2)
+    }
+
+    public fun g2_sub(e1: &Element<G2>, e2: &Element<G2>): Element<G2> {
+        group_ops::sub(G2_TYPE, e1, e2)
+    }
+
+    public fun g2_mul(e1: &Element<Scalar>, e2: &Element<G2>): Element<G2> {
+        group_ops::mul(G2_TYPE, e1, e2)
+    }
+
+    // Returns e2 / e1, fails if scalar is zero.
+    public fun g2_div(e1: &Element<Scalar>, e2: &Element<G2>): Element<G2> {
+        group_ops::div(G2_TYPE, e1, e2)
+    }
+
+    public fun g2_neg(e: &Element<G2>): Element<G2> {
+        g2_sub(&g2_identity(), e)
+    }
+
+    // Hash using DST = BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_
+    public fun hash_to_g2(m: &vector<u8>): Element<G2> {
+        group_ops::hash_to(G2_TYPE, m)
+    }
+
+    // Let 'scalars' be the vector [s1, s2, ..., sn] and 'elements' be the vector [e1, e2, ..., en].
+    // Returns s1*e1 + s2*e2 + ... + sn*en.
+    public fun g2_multi_scalar_multiplication(scalars: &vector<Element<Scalar>>, elements: &vector<Element<G2>>): Element<G2> {
+        group_ops::multi_scalar_multiplication(G2_TYPE, scalars, elements)
+    }
+
+    /////////////////////////////////
+    ////// Gt group operations //////
+
+    // TODO: remove? not sure I see a use case for this
```

Comment:
@kchalkias @joyqvq @jonas-lj do you use any use case for `from_bytes` to `GT` in Move?
`to_bytes` may be used for generating an input to a random oracle, or KDF.

Reply:
removed